### PR TITLE
Build docker image once during build

### LINKED ISSUE #579: Build docker images once during build flow
It's built 2+ times in more than 1 workflow.

### PR DESCRIPTION
**Pull request by itaiad200**
closes #579

### CODE REVIEW

File: .github/workflows/docker-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Extract deploy version
+        shell: bash
+        run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
+        id: release-version
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Tag Amazon ECR
```
Comment (arielshaqed) [inline .github/workflows/docker-publish.yaml:35]:
```suggestion
      - name: Tag on AWS ECR and on DockerHub
```

**Review by nopcoder — CHANGES_REQUESTED** [review]
1. I think we would like to create an action to build the docker and use it in both flows
2. Prefer not to 'deploy' our dockers to docker hub while running system test - want to verify the PR without talking to a public repository

**Review by arielshaqed — APPROVED** [review]
Cool!
How much faster does it run?  (Because sometimes the various hubs can be... slow...).

File: .env [hunk 2]
```diff
@@ -1,2 +1,3 @@
 TAG=latest
-LAKEFS_BLOCKSTORE_TYPE=local
\ No newline at end of file
+LAKEFS_BLOCKSTORE_TYPE=local
+DOCKER_REG=https://hub.docker.com
```
Comment (nopcoder) [inline .env:3]: we don't need this one - running lakefs doesn't use it and the name treeverse/lakefs is always docker hub. this is the default. if not you need to specify the address of your repository.

File: docker-compose.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   lakefs:
-    image: "treeverse/lakefs:${TAG}"
+    image: "${DOCKER_REG}/treeverse/lakefs:${TAG}"
```
Comment (nopcoder) [inline docker-compose.yaml:4]: did we reach the point when we like docker compose for the end-user and one for the test system?

File: .env (hunk 2 above)
Comment (itaiad200) [reply .env:3]: How else would I pull images that exist only in ECR?
Remember that when Nessie pulls images, it's not pushed to docker hub yet.

**Comment by itaiad200** [comment]
> Cool!
> How much faster does it run? (Because sometimes the various hubs can be... slow...).

Since the workflows are running in parallel, it doesn't really save us time just yet. We'll benefit later when the images are used by the benchmark tests.

File: docker-compose.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
   lakefs:
-    image: "treeverse/lakefs:${TAG}"
+    image: treeverse/lakefs:${TAG}"
```
Comment (nopcoder) [inline docker-compose.yaml:4]: think you are missing or have extra "

File: docker-compose.yaml [hunk 5]
```diff
@@ -15,7 +15,6 @@ services:
       - LAKEFS_BLOCKSTORE_S3_CREDENTIALS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
       - LAKEFS_BLOCKSTORE_S3_CREDENTIALS_ACCESS_SECRET_KEY=${AWS_SECRET_ACCESS_KEY}
       - LAKEFS_LOGGING_LEVEL
-      - LAKEFS_BLOCKSTORE_GS_CREDENTIALS_JSON
```
Comment (nopcoder) [inline docker-compose.yaml:18]: just making sure we miss a merge with master and we are not taking this one out

File: nessie/ops/.env [hunk 6]
```diff
@@ -0,0 +1,3 @@
+TAG=latest
+LAKEFS_BLOCKSTORE_TYPE=local
+DOCKER_REG=977611293394.dkr.ecr.us-east-1.amazonaws.com
```
Comment (nopcoder) [inline nessie/ops/.env:3]: Not part of this PR but I think we should look into having all this code and ops as code in a private repository as this code / test env is not public and managed on the public repository.

File: docker-compose.yaml (hunk 5 above)
Comment (itaiad200) [reply docker-compose.yaml:18]: Why do we need this in the global `docker-compose`? It can be added by the user if GCS is needed, but our quickstart refers to local/S3 storage.

File: nessie/ops/.env (hunk 6 above)
Comment (itaiad200) [reply nessie/ops/.env:3]: Managing ops in a private repo has many disadvantages, let's discuss offline.

**Review by nopcoder — APPROVED** [review]

**Review by nopcoder — APPROVED** [review]

File: nessie/ops/.env (hunk 6 above)
Comment (arielshaqed) [reply nessie/ops/.env:3]: Or (ab-)use "secret"s to store our specific information (like AWS account ID).
